lib/ansible/modules/files/yedit.py

```diff
@@ -0,0 +1,965 @@
+    - A list of edits to perform.  These follow the same format as a single edit
+    required: false
+    aliases: []
+  value_type:
+    description:
+    - The python type of the incoming value.
+    required: false
+    default: ''
+    aliases: []
+  update:
+    description:
+    - Whether the update should be performed on a dict/hash or list/array
+    - object.
+    required: false
+    default: false
+    aliases: []
+    type: bool
+  append:
+    description:
+    - Whether to append to an array/list. When the key does not exist or is
+    - null, a new array is created. When the key is of a non-list type,
+    - nothing is done.
+    required: false
+    default: false
+    aliases: []
+    type: bool
+  index:
+    description:
+    - Used in conjunction with the update parameter.  This will update a
+    - specific index in an array/list.
+    required: false
+    aliases: []
+  curr_value:
+    description:
+    - Used in conjunction with the update parameter.  This is the current
+    - value of 'key' in the yaml file.
+    required: false
+    default: None
+    aliases: []
+  curr_value_format:
+    description:
+    - Format of the incoming current value.
+    choices: ["yaml", "json", "str"]
+    required: false
+    default: yaml
+    aliases: []
+  backup_ext:
+    description:
+    - The backup file's appended string.
+    required: false
+    aliases: []
+  backup:
+    description:
+    - Whether to make a backup copy of the current file when performing an
+    - edit.
+    required: false
+    default: false
+    aliases: []
+    type: bool
+  separator:
+    description:
+    - The separator being used when parsing strings.
+    required: false
+    default: '.'
+    aliases: []
+author:
+- "Kenny Woodson <kwoodson@redhat.com>"
+extends_documentation_fragment: []
+'''
+
+EXAMPLES = '''
+# Simple insert of key, value
+- name: insert simple key, value
+  yedit:
+    src: somefile.yml
+    key: test
+    value: somevalue
+    state: present
+# Results:
+# test: somevalue
+
+# Multilevel insert of key, value
+- name: insert simple key, value
+  yedit:
+    src: somefile.yml
+    key: a#b#c
+    value: d
+    state: present
+# Results:
+# a:
+#   b:
+#     c: d
+#
+# multiple edits at the same time
+- name: perform multiple edits
+  yedit:
+    src: somefile.yml
+    edits:
+    - key: a#b#c
+      value: d
+    - key: a#b#c#d
+      value: e
+    state: present
+# Results:
+# a:
+#   b:
+#     c:
+#       d: e
+'''
+
+import copy  # noqa: F401
+import fcntl  # noqa: F401
+import json   # noqa: F401
+import os  # noqa: F401
+import re  # noqa: F401
+import shutil  # noqa: F401
+import time  # noqa: F401
+
+try:
+    import ruamel.yaml as yaml  # noqa: F401
+except ImportError:
+    import yaml  # noqa: F401
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+class YeditException(Exception):
+    ''' Exception class for Yedit '''
+    pass
+
+
+# pylint: disable=too-many-public-methods,too-many-instance-attributes
```

Comment:
`RETURN` block missing

Reply:
I'm not sure what this returns to.  I'll look at some other modules to determine what this refers to.

Reply:
Updated.